### PR TITLE
HAAR-1099 re-factor NomisApiService

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        "email" to centralAdminUser.email,
+        "firstName" to centralAdminUser.firstName,
+        "lastName" to centralAdminUser.lastName
+      ),
+      NomisUserDetails::class.java
+    )
+  }
+
+  fun createGeneralUser(generalUser: CreateUserRequest): NomisUserDetails {
+    RolesApiService.log.debug("Create DPS general user - {}", generalUser.username)
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt:34]: same as above..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt [hunk 2]
```diff
@@ -128,52 +124,6 @@ class HmppsManageUsersApiExceptionHandler {
       .body(ErrorResponse(status = BAD_REQUEST.value(), developerMessage = e.message))
   }
 
-  @ExceptionHandler(RoleNotFoundException::class)
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt:131]: just make sure these are add as a part of auth refactor and none of manager user UI relies on these error codes

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt [hunk 3]
```diff
@@ -9,25 +9,23 @@ import uk.gov.justice.digital.hmpps.manageusersapi.resource.RoleDescriptionAmend
 import uk.gov.justice.digital.hmpps.manageusersapi.resource.RoleNameAmendment
 import uk.gov.justice.digital.hmpps.manageusersapi.service.AdminType.DPS_ADM
 import uk.gov.justice.digital.hmpps.manageusersapi.service.AdminType.DPS_LSA
-import uk.gov.justice.digital.hmpps.manageusersapi.service.nomis.NomisApiService
 
 @Service
 class RolesService(
-  val nomisApiService: NomisApiService,
-  val rolesApiService: RolesApiService
+  val nomisRolesApiService: uk.gov.justice.digital.hmpps.manageusersapi.adapter.nomis.RolesApiService,
```
[the remark this answers is not in the document]
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesService.kt]: Good spot - have changed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+@Service(value = "nomisUserApiService")
+class UserApiService(
+  @Qualifier("nomisWebClientUtils") val nomisWebClientUtils: WebClientUtils
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun createCentralAdminUser(centralAdminUser: CreateUserRequest): NomisUserDetails {
+    RolesApiService.log.debug("Create DPS central admin user - {}", centralAdminUser.username)
```
[the remark this answers is not in the document]
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt (hunk 2 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/ManageUsersApiExceptionHandler.kt:131]: Done

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**